### ISSUE TITLE
requirements.txt install error

### ISSUE THREAD

**Issue by sdancer75** [volatilityfoundation/volatility3#901]
**Describe the bug**
Its not actually a real volatility3 bug, but it seems python-snappy is not supported with latest version of python 3.11. So, its wise to give detailed explanation since you write that min requirements is Python 3.7 or greater.

**Context**
Volatility Version:  3 framework 2.4.1
Operating System:  win10 pro
Python Version:  3.11
Suspected Operating System:  
Command:  

**To Reproduce**

1. Use command '...'D:\Python311\Scripts\pip3 install -r requirements.txt
2. See error

```
Collecting pefile>=2017.8.1
  Using cached pefile-2022.5.30.tar.gz (72 kB)
  Preparing metadata (setup.py) ... done
Requirement already satisfied: yara-python>=3.8.0 in d:\python311\lib\site-packages (from -r requirements.txt (line 8)) (4.2.3)
Collecting capstone>=3.0.5
  Using cached capstone-4.0.2-py2.py3-none-win_amd64.whl (896 kB)
Collecting pycryptodome
  Using cached pycryptodome-3.16.0-cp35-abi3-win_amd64.whl (1.7 MB)
Collecting leechcorepyc>=2.4.0
  Using cached leechcorepyc-2.14.1-cp36-abi3-win_amd64.whl (357 kB)
Collecting python-snappy==0.6.0
  Using cached python-snappy-0.6.0.tar.gz (21 kB)
  Preparing metadata (setup.py) ... done
Collecting future
  Using cached future-0.18.3.tar.gz (840 kB)
  Preparing metadata (setup.py) ... done
Installing collected packages: python-snappy, pycryptodome, leechcorepyc, future, capstone, pefile
  DEPRECATION: python-snappy is being installed using the legacy 'setup.py install' method, because it does not have a 'pyproject.toml' and the 'wheel' package is not installed. pip 23.1 will enforce this behaviour change. A possible replacement is to enable the '--use-pep517' option. Discussion can be found at https://github.com/pypa/pip/issues/8559
  Running setup.py install for python-snappy ... error
  error: subprocess-exited-with-error

  × Running setup.py install for python-snappy did not run successfully.
  │ exit code: 1
  ╰─> [28 lines of output]
      D:\Python311\Lib\site-packages\setuptools\_distutils\dist.py:264: UserWarning: Unknown distribution option: 'cffi_modules'
        warnings.warn(msg)
      running install
      D:\Python311\Lib\site-packages\setuptools\command\install.py:34: SetuptoolsDeprecationWarning: setup.py install is deprecated. Use build and pip and other standards-based tools.
        warnings.warn(
      running build
      running build_py
      creating build
      creating build\lib.win-amd64-cpython-311
      creating build\lib.win-amd64-cpython-311\snappy
      copying snappy\hadoop_snappy.py -> build\lib.win-amd64-cpython-311\snappy
      copying snappy\snappy.py -> build\lib.win-amd64-cpython-311\snappy
      copying snappy\snappy_cffi.py -> build\lib.win-amd64-cpython-311\snappy
      copying snappy\snappy_cffi_builder.py -> build\lib.win-amd64-cpython-311\snappy
      copying snappy\snappy_formats.py -> build\lib.win-amd64-cpython-311\snappy
      copying snappy\__init__.py -> build\lib.win-amd64-cpython-311\snappy
      copying snappy\__main__.py -> build\lib.win-amd64-cpython-311\snappy
      running build_ext
      building 'snappy._snappy' extension
      creating build\temp.win-amd64-cpython-311
      creating build\temp.win-amd64-cpython-311\Release
      creating build\temp.win-amd64-cpython-311\Release\snappy
      "d:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\VC\Tools\MSVC\14.16.27023\bin\HostX86\x64\cl.exe" /c /nologo /O2 /W3 /GL /DNDEBUG /MD -ID:\Python311\include -ID:\Python311\Include "-Id:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\VC\Tools\MSVC\14.16.27023\ATLMFC\include" "-Id:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\VC\Tools\MSVC\14.16.27023\include" "-IC:\Program Files (x86)\Windows Kits\NETFXSDK\4.6.1\include\um" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\ucrt" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\shared" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\um" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\winrt" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\cppwinrt" /Tcsnappy/crc32c.c /Fobuild\temp.win-amd64-cpython-311\Release\snappy/crc32c.obj
      crc32c.c
      "d:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\VC\Tools\MSVC\14.16.27023\bin\HostX86\x64\cl.exe" /c /nologo /O2 /W3 /GL /DNDEBUG /MD -ID:\Python311\include -ID:\Python311\Include "-Id:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\VC\Tools\MSVC\14.16.27023\ATLMFC\include" "-Id:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\VC\Tools\MSVC\14.16.27023\include" "-IC:\Program Files (x86)\Windows Kits\NETFXSDK\4.6.1\include\um" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\ucrt" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\shared" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\um" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\winrt" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\cppwinrt" /EHsc /Tpsnappy/snappymodule.cc /Fobuild\temp.win-amd64-cpython-311\Release\snappy/snappymodule.obj
      snappymodule.cc
      snappy/snappymodule.cc(32): fatal error C1083: Cannot open include file: 'snappy-c.h': No such file or directory
      error: command 'd:\\Program Files (x86)\\Microsoft Visual Studio\\2017\\Enterprise\\VC\\Tools\\MSVC\\14.16.27023\\bin\\HostX86\\x64\\cl.exe' failed with exit code 2
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: legacy-install-failure

× Encountered error while trying to install package.
╰─> python-snappy

note: This is an issue with the package mentioned above, not pip.
hint: See above for output from the failure.
```

**Expected behavior**

**Example output**
Please copy and paste the text demonstrating the issue, ideally with verbose output turned on (`vol.py -vvv ...`).  

Text is preferred to screenshots for searching and to talk about specific parts of the output.

**Additional information**
Add any other information about the problem here.

**Comment by eve-mem** [comment]
Hello, this sounds very similar to issue 804. Have a read of that and see if it helps fix your problem. 

https://github.com/volatilityfoundation/volatility3/issues/804

**Comment by sdancer75** [comment]
Hi, thank you for your answer. As far as I can understand this is a python-snappy problem that does not support the latest version of python.  So, the instructions should be changed accordingly.

**Comment by ikelos** [comment]
It's a quirk of the way python-snappy is packaged.  I believe it does work, but requires either the library to have already been installed separately, or to have the headers and suitable compiler available for it to be built.

There is scope to move to https://github.com/ethereum/py-snappy which is a pure python version of the library, but reimplementing and testing support for a single file format (AVML, which is currently the only component that requires snappy support) is fairly low on the priority list.

**Comment by sdancer75** [comment]
Yes, I tried to install it locally but also failed. So I gave up. I wanted just to make some tests, not very important though.

**Comment by fukusuket** [comment]
I had the same issue in my environment(on Windows10 Pro, Python 3.10.7).

The installation error has been resolved in my environment by changing the following `python-snappy` version as follows:
https://github.com/volatilityfoundation/volatility3/blob/v2.4.0/requirements.txt#L22
`python-snappy==0.6.0` to `python-snappy==0.6.1`
(But I have not confirmed whether the  all function works with version `0.6.1`)

I checked the PyPI repository of `python-snappy` and found the following:

In `v0.6.0` there is no distribution Windows - Python3.10  as follows.
https://pypi.org/project/python-snappy/0.6.0/#files

On the other hand, in `v0.6.1` there is a distribution Windows - Python3.10 as follows.
https://pypi.org/project/python-snappy/0.6.1/#files

Due to the above situation, `python-snappy v0.6.0`  seems to require C compilation(on Windows+Python3.10).
